silverback-integration-kafka/src/Silverback.Integration.Kafka/Messaging/KeyHelper.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json;
+using Silverback.Messaging.Messages;
+
+namespace Silverback.Messaging
+{
+    /// <summary>
+    /// Attribute for decorating the message properties.
+    /// </summary>
+    /// <seealso cref="System.Attribute" />
+    [AttributeUsage(AttributeTargets.Property)]
+    public class KeyMember : Attribute
+    {
+
+    }
+
+    /// <summary>
+    /// Helps to retrieve the message properties with KeyMember attribute for creating a key.
+    /// </summary>
+    internal static class KeyHelper
+    {
+        public static byte[] GetMessageKey(IIntegrationMessage message)
+        {
+            var keyObj = message.GetType().GetProperties().Where(p => p.IsDefined(typeof(KeyMember), false))
+                .ToDictionary(p => p.Name, p => p.GetValue(message, null));
+
+            return keyObj.Count > 0
+                ? Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(keyObj))
```

Comment:
Is this ok like this? The size is not important?
Is it not worth to create an hash or something? Or would it be counterproductive?